### PR TITLE
Fix/do not use role contract if notary disabled

### LINKED ISSUE #672: Inner Ring and Storage nodes should not use side chain role management contract if notary disabled
Side chain role management contract stores list of Inner Ring public keys in `NeoFSAlphabet` role. However we can't manage role management contract without notary service that can create transaction signed by side chain committee. Therefore if notary service is disabled then Netmap contract used as storage of this list. 

However Inner Ring nodes and Storage nodes use role management contract to get list of Inner Ring nodes anyway which is incorrect.

### PR DESCRIPTION
**Pull request by carpawell**
Closes #672.

### CODE REVIEW

File: pkg/morph/client/netmap/innerring.go [hunk 1]
```diff
@@ -9,5 +16,62 @@ func (c *Client) SetInnerRing(keys keys.PublicKeys) error {
...
+		c.innerRingList,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not perform test invocation (%s): %w", c.innerRingList, err)
+	}
+
+	return irsKeysFromStackItem(prms, c.innerRingList)
+}
+
+func irsKeysFromStackItem(stack []stackitem.Item, method string) (keys.PublicKeys, error) {
```
Comment (alexvanin) [inline pkg/morph/client/netmap/innerring.go:35]: irs?

**Review by alexvanin — COMMENTED** [review]
What about `NeoFSAlphabetList()` invocation in `indexer.go`?

File: pkg/morph/client/netmap/innerring.go (hunk 1 above)
Comment (carpawell) [reply pkg/morph/client/netmap/innerring.go]: fixed

**Review by carpawell — COMMENTED** [review]
@alexvanin, forget about IR indexer, added in https://github.com/nspcc-dev/neofs-node/pull/705/commits/f481da6a302b30b7119364e6c2419b3d0ad2c6c8

File: pkg/morph/client/netmap/innerring.go (hunk 1 above)
Comment (carpawell) [reply pkg/morph/client/netmap/innerring.go]: fixed

File: pkg/innerring/innerring.go [hunk 2]
```diff
@@ -151,6 +151,26 @@ var (
...
+	return fN.cli.NeoFSAlphabetList()
+}
+
+type irFetcherWithoutNotary struct {
+	nm *nmWrapper.Wrapper
+}
+
+// InnerRingKeys fetches list of innerring keys from netmap contract
+// in side chain.
+func (f irFetcherWithoutNotary) InnerRingKeys() (keys.PublicKeys, error) {
```
Comment (alexvanin) [inline pkg/innerring/innerring.go:170]: Move all fetcher definitions to `indexer.go`.

File: pkg/innerring/processors/governance/process_update.go [hunk 3]
```diff
@@ -53,12 +53,24 @@ func (gp *Processor) processAlphabetSync() {
...
-	// 2. Update NeoFSAlphabet role in side chain.
-	innerRing, err := gp.morphClient.NeoFSAlphabetList()
-	if err != nil {
-		gp.log.Error("can't fetch inner ring list from side chain",
-			zap.String("error", err.Error()))
+	// 2. Update NeoFSAlphabet role/netmap contract in side chain.
+	var innerRing keys.PublicKeys
+
+	if gp.morphClient.ProbeNotary() {
+		innerRing, err = gp.morphClient.NeoFSAlphabetList()
```
Comment (alexvanin) [inline pkg/innerring/processors/governance/process_update.go:60]: I suggest to define irFetcher interface in the governance processor and reuse implementation from indexer.

File: pkg/innerring/innerring.go (hunk 2 above)
Comment (carpawell) [reply pkg/innerring/innerring.go]: https://github.com/nspcc-dev/neofs-node/pull/705/commits/40a6ea7caf67e04d48d9df58103d259c90bd369d

File: pkg/innerring/processors/governance/process_update.go (hunk 3 above)
Comment (carpawell) [reply pkg/innerring/processors/governance/process_update.go]: https://github.com/nspcc-dev/neofs-node/pull/705/commits/0572d3becaa26f475cff9fb6b17c5de9e9082042

**Review by alexvanin — APPROVED** [review]

**Pull request merged**